packages/fuels-abigen-macro/src/attributes.rs

```diff
@@ -0,0 +1,262 @@
+use itertools::Itertools;
+use proc_macro2::Span;
+use syn::{
+    parenthesized, parse::ParseStream, parse_macro_input::ParseMacroInput, AttributeArgs, Error,
+    Lit, LitStr, Meta, MetaNameValue, NestedMeta,
+};
+
+// Used to parse the attributes inside the parentheses of a Contract, Script or
+// Predicate command in the `abigen!` e.g. `abigen!(Contract(will_parse = "this_here"))`
+#[derive(Debug)]
+pub(crate) struct Attributes {
+    content_span: Span,
+    attrs: Vec<MetaNameValue>,
+}
+
+impl Attributes {
+    pub(crate) fn new(input: ParseStream, valid_attr_names: &[&str]) -> syn::Result<Self> {
+        let content;
+        parenthesized!(content in input);
+
+        let content_span = content.span();
+
+        let attrs = Self::extract_attrs(&content)?;
+        Self::validate_attrs(&attrs, valid_attr_names)?;
+
+        Ok(Self {
+            content_span,
+            attrs,
+        })
+    }
+
+    pub(crate) fn get_as_lit_str(&self, attr_name: &str) -> syn::Result<LitStr> {
```

Comment:
If it were a simple getter, then yes. 

It resembles the interface of `HashMap`, so that's where the `get` part is coming from.

`as_lit_str` could mean 'get the contents as a literal string', but there are multiple name-value pairs in the contents, so that doesn't exactly fit. 

But then again I wanted this get to be a 'get-or-fail`, not a option-returning get as `HashMap` does it, so I renamed the ordinary `get` to `try_get` trying to communicate that it is okay there to not succeed, but not here.